ui/tests/e2e/service-mesh-performance.spec.js

```diff
@@ -0,0 +1,90 @@
+      await page.getByLabel('Add Performance Profile').click();
+      await page.getByLabel('Profile Name').fill(profileName);
+      await page.getByLabel('Service Mesh').click();
+      await page.getByRole('option', { name: serviceMesh }).click();
+      await page.getByLabel('URL to test').fill(url);
+      await page.getByLabel('Concurrent requests').fill('2');
+      await page.getByLabel('Queries per second').fill('2');
+      await page.getByLabel('Duration').fill('15s');
+      await page.getByLabel(loadGenerator).check();
+
+      const runPerformanceTest = await page.getByTestId('run-performance-test');
+      await expect(runPerformanceTest).toBeVisible();
+      await runPerformanceTest.click();
+      await expect(page.getByText('Load test has been submitted')).toBeVisible();
+    });
+
+    test(`View detailed result of a performance profile (Graph Visualiser) with load generator "${loadGenerator}" and service mesh "${serviceMesh}"`, async ({
+      page,
+    }) => {
+      await page.goto(`${ENV.MESHERY_SERVER_URL}/performance/profiles`);
+      await page.waitForSelector(`text=${profileName}`, {
+        state: 'visible',
+      });
+      await page.getByRole('button', { name: 'View Results', exact: true }).first().click();
+
+      await page.getByTestId('open-performance-result-bar-chart').first().click();
+      await expect(page.locator('[data-test="sentinelStart"]')).toBeHidden();
+      await expect(await page.getByText(`URL: ${url}`, { exact: true })).toBeVisible();
+    });
+
+    test(`Edit the configuration of a performance profile with load generator "${loadGenerator}" and service mesh "${serviceMesh}"`, async ({
+      page,
+    }) => {
+      await page.goto(`${ENV.MESHERY_SERVER_URL}/performance/profiles`);
+      await page.getByText(profileName, { exact: true }).first().click();
+
+      await page.getByTestId('performanceProfileCard-edit').click();
+      await page.getByLabel('Service Mesh').click();
+      await page.getByRole('option', { name: serviceMesh }).click();
+      await page.getByLabel('Concurrent requests').fill('3');
+      await page.getByLabel(loadGenerator).check();
+
+      const runPerformanceTest = await page.getByTestId('run-performance-test');
+      await expect(runPerformanceTest).toBeEnabled();
+      await runPerformanceTest.click();
+      await expect(page.getByText('Load test has been submitted')).toBeVisible();
+    });
+
+    test(`Compare test of a performance profile with load generator "${loadGenerator}" and service mesh "${serviceMesh}"`, async ({
+      page,
+    }) => {
+      await page.goto(`${ENV.MESHERY_SERVER_URL}/performance/profiles`);
+      await page.getByText(profileName, { exact: true }).first().click();
+      await page.getByRole('button', { name: 'View Results', exact: true }).first().click();
+      await page.locator('[data-testid="MUIDataTableBodyRow-0"] input[type="checkbox"]').check();
```

Comment:
Can we use playwright locator here `page.getByTestId`?

Reply:
@jerensl do u mean something like this 
await page.getByTestId('MUIDataTableBodyRow-0').locator('input[type="checkbox"]').check();
?
or should i inject testid ,i tried but couldnt find where to inject.

Reply:
@ayushsawant464 MUI test id is fine